src/time_travel/patchers/select_patcher.py

```diff
@@ -23,42 +26,74 @@ def __init__(self, *args, **kwargs):
         
         self.patches = [mock.patch('select.select', self.select)]
         
+    @classmethod
+    def register_events_types(cls, main_events_types):
+        """Register select events types."""
+        setattr(main_events_types, 'select', cls.EventTypes) 
```

Comment:
I think it's a bit crooked. Why does the patcher need to know how to set things in the library? You're gonna use entry points, that means that the patcher should tell the library how to register itself to it, not the other way around.
The patchers should expose an interface that will return the event types, and the main library needs to do all the registration business.